packages/cosmic-swingset/lib/ag-solo/vats/rendezvous.js

```diff
@@ -0,0 +1,277 @@
+// @ts-check
+import makeStore from '@agoric/store';
+
+import { assert, details } from '@agoric/assert';
+import { makePromiseKit } from '@agoric/promise-kit';
+
+/**
+ * @typedef {string} PeerAddress the string representation of an Agoric peer's
+ * identity.  For Cosmos, this will be the Agoric account ID, like
+ * `agoric16vcjpzjr8uhvwwngyzkpcwe4x0qmnjdu63hc9u`.  For the simulated chain,
+ * this will be something like `mySimGCI-client`.
+ */
+
+/**
+ * @typedef {PeerAddress | Array<PeerAddress>} PeerAddresses one or many peer
+ * addresses
+ */
+
+const RENDEZVOUS_ALREADY = harden({
+  toString() {
+    return 'RENDEZVOUS_ALREADY';
+  },
+});
+
+/**
+ * @typedef {Object} RendezvousPrivateService A closely-held service associated
+ * with a specific Agoric peer.  It allows you to exchange values (possibly
+ * ocaps) with another peer, given a third-party that can exchange the other
+ * side's address strings between you.
+ *
+ * Consider the Agoric chain, A, and two ag-solos attached to the chain: B and
```

Comment:
@katelynsills here is the more detailed explanation you were asking for.  Once we're not consumed by a deadline, I'd love it if you can do another review.